Source/Dafny/Triggers/QuantifiersCollection.cs

```diff
@@ -299,32 +307,42 @@ internal class QuantifierGroup {
         }
 
         AddTriggersToMessage("Selected triggers:", q.Candidates, msg, indent);
+        if (q.Candidates.Count > 0) {
+          SetRelatedLocationMessage("Trigger position:");
+        }
         AddTriggersToMessage("Rejected triggers:", q.RejectedCandidates, msg, indent, true);
+        if (q.RejectedCandidates.Count > 0 && q.Candidates.Count == 0) {
+          SetRelatedLocationMessage("Rejected trigger position:");
+        }
 
 #if QUANTIFIER_WARNINGS
-        var WARN_TAG = DafnyOptions.O.UnicodeOutput ? "⚠ " : "/!\\ ";
+        var WARN_TAG = DafnyOptions.O.UnicodeOutput ? "⚠ " : @"/!\ ";
         var WARN_TAG_OVERRIDE = suppressWarnings ? "(Suppressed warning) " : WARN_TAG;
         var WARN_LEVEL = suppressWarnings ? ErrorLevel.Info : ErrorLevel.Warning;
         var WARN = indent + WARN_TAG_OVERRIDE;
         if (!q.CandidateTerms.Any()) {
           errorLevel = WARN_LEVEL;
           msg.Append(WARN).AppendLine("No terms found to trigger on.");
+          SetRelatedLocationMessage("Potential trigger not suitable:");
         } else if (!q.Candidates.Any()) {
           errorLevel = WARN_LEVEL;
           msg.Append(WARN).AppendLine("No trigger covering all quantified variables found.");
+          SetRelatedLocationMessage("Potential trigger not suitable:");
         } else if (!q.CouldSuppressLoops && !q.AllowsLoops) {
           errorLevel = WARN_LEVEL;
           msg.Append(WARN).AppendLine("Suppressing loops would leave this expression without triggers.");
+          SetRelatedLocationMessage("Trigger position:");
         } else if (suppressWarnings) {
           errorLevel = ErrorLevel.Warning;
           msg.Append(indent).Append(WARN_TAG).AppendLine("There is no warning here to suppress.");
+          SetRelatedLocationMessage("Trigger position:");
```

Comment:
Why do you add a location message to a warning about `nowarn`?

Reply:
I am not adding it, it was already there, I'm just replacing the "Related location" message with something more explicit. The alternative would be to just ignore it. What do you think?

Reply:
This should probably be in an else branch:

```
else {
  if (suppressWarnings) { … }
  SetRelatedLocation…
}